internal/mode/static/nginx/config/generator.go

```diff
@@ -177,3 +215,98 @@ func generateCertBundle(id dataplane.CertBundleID, cert []byte) file.File {
 func generateCertBundleFileName(id dataplane.CertBundleID) string {
 	return filepath.Join(secretsFolder, string(id)+".crt")
 }
+
+type mgmtConf struct {
+	Endpoint            string
+	Resolver            string
+	DeploymentCtxFile   string
+	SkipVerify          bool
+	CACertExists        bool
+	ClientSSLCertExists bool
+}
+
+// generateMgmtFile generates the NGINX Plus configuration file for the mgmt block. As part of this,
+// it writes the deployment context file that is referenced in the mgmt block.
+func (g GeneratorImpl) generateMgmtFiles() []file.File {
+	var files []file.File
+
+	podNSName := types.NamespacedName{
+		Namespace: g.podConfig.Namespace,
+		Name:      g.podConfig.Name,
+	}
+
+	conf := mgmtConf{
+		Endpoint:            g.usageReportConfig.Endpoint,
+		Resolver:            g.usageReportConfig.Resolver,
+		SkipVerify:          g.usageReportConfig.SkipVerify,
+		CACertExists:        g.usageReportConfig.CASecretName != "",
+		ClientSSLCertExists: g.usageReportConfig.ClientSSLSecretName != "",
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	data, err := getDeploymentCtx(ctx, g.k8sClientReader, podNSName)
```

Comment:
Why get the deployment context here instead of in the state package? This adds a dependency on the k8s API to the generator.

Reply:
Do you think it's okay to store the deployment context in the dataplane Configuration object? Or is that too unrelated to what the config contains right now?

Reply:
I think it's ok. It's pretty much just metadata about the cluster.